aana/api/request_handler.py

```diff
@@ -92,30 +95,46 @@ async def is_ready(self):
         """
         return AanaJSONResponse(content={"ready": self.ready})
 
-    async def call_endpoint(self, path: str, **kwargs: dict[str, Any]) -> Any:
-        """Call the endpoint from FastAPI with the given name.
+    async def execute_task(self, task_id: str) -> Any:
+        """Execute a task.
 
         Args:
-            path (str): The path of the endpoint.
-            **kwargs: The arguments to pass to the endpoint.
+            task_id (str): The task ID.
 
         Returns:
             Any: The response from the endpoint.
         """
-        for e in self.endpoints:
-            if e.path == path:
-                endpoint = e
-                break
-        else:
-            raise ValueError(f"Endpoint {path} not found")  # noqa: TRY003
-
-        if not endpoint.initialized:
-            await endpoint.initialize()
-
-        if endpoint.is_streaming_response():
-            return [item async for item in endpoint.run(**kwargs)]
+        session = get_session()
```

Comment:
> can we set session as a attr of the class

We need to change that later. We cannot reuse the seesion. Session cannot be shared among concurrent threads or asyncio tasks without careful synchronization. 

Right now it kinda works because we are using syncronious session so it just blocks async threads but this is not a correct way.

See https://docs.sqlalchemy.org/en/20/orm/session_basics.html#is-the-session-thread-safe-is-asyncsession-safe-to-share-in-concurrent-tasks